namer/consul/src/main/scala/io/buoyant/namer/consul/SvcAddr.scala

```diff
@@ -80,6 +82,19 @@ private[consul] object SvcAddr {
             )
             stopped = true
             Future.Unit
+            // this exception case checks if we queried for a service in a datacenter that
+            // doesn't exist. We capture this case so that we can return Addr.Neg to prevent
+            // service name resolution from timing out.
+          case Throw(e: UnexpectedResponse) if e.rsp.contentString == DatacenterErrorMessage =>
+            log.log(
+              failureLogLevel,
+              s"consul datacenter $datacenter service ${key.name} " +
```

Comment:
This isn't a timeout, right?  it's just the consul API reporting that the requested DC does not exist.  I think we can improve the error message here to reflect that.